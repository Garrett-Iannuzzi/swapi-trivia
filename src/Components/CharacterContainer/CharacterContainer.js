import React from 'react';
import './CharacterContainer.scss';
import CharacterCard from '../CharacterCard/CharacterCard.js';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CharacterContainer = ( { characters, resetCharacters }) => {
  const characterInfo = characters.map(character => {
    return(
      <CharacterCard
        {...character}
        key={character.character}
      />
    )
  })
  return(
    <main className="character-container">
      <Link to="/movies">
        <button className="button-movies" onClick={() => resetCharacters()}>Return to Movies</button>
      </Link>
      <p className="opening-crawl">{characters[0].openingCrawl}</p>
      <section className="section-character-container">
        {characterInfo}
      </section>
    </main>
  )
}

export default CharacterContainer;

CharacterContainer.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object),
  resetCharacters: PropTypes.func,
}
