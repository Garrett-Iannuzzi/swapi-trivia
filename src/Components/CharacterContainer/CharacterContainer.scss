.character-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 81vh;
}

.opening-crawl {
  width: 75%;
}

.section-character-container {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 82vh;
  justify-content: space-around;
  padding: 10px;
  overflow: scroll;
  width: 95%;
}
