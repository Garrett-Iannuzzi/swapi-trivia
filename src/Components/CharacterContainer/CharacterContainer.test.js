import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CharacterContainer from '../CharacterContainer/CharacterContainer.js';

describe('CharacterContainer', () => {

  let wrapper;
  const resetCharactersMock = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<CharacterContainer
      characters={[{
        character: "C-3PO",
        creature: "artificial",
        name: "Tatooine",
        openingCrawl: "the guardians of",
        population: "200000",
        relatedFilms:[
          {relatedFilms: "The Empire Strikes Back"},
          {relatedFilms: "Attack of the Clones"}],
          species: "Droid"
      }, {
        character: "R2-D2",
        creature: "artificial",
        name: "Naboo",
        openingCrawl: "Turmoil has engulfed",
        population: "4500000000",
        relatedFilms: [
          {relatedFilms: "The Empire Strikes Back"},
          {relatedFilms: "Attack of the Clones"}],
        species: "Droid"
      }]}
      resetCharacters={resetCharactersMock}
      />);
  })

  it('Should match the snapshot with all the proper data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
