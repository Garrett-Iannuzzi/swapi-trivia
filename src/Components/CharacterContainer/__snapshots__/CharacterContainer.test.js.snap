// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer Should match the snapshot with all the proper data passed in 1`] = `ShallowWrapper {}`;
